case_studies/adaptive_tiling/region_encoder.py

```diff
@@ -151,24 +121,83 @@ def variational_mode(
                     (i_above, i_above, i_below, i_below),
                     (j_left, j_right, j_left, j_right),
                 )
-                if aux_var[b, i, j] == aux_var[c_idx].max():
-                    tile_is_on_array[c_idx] = 0  # turn off other tiles
-                    tile_is_on_array[b, i, j] = 1
+                if d["aux_var"][b, i, j] == d["aux_var"][c_idx].max():
```

Comment:
This call to `max` looks suspicious, but I don't have a good understanding of what is going on here. In a probabilistic catalog I think you'd only check `max` if you're trying to form a map estimate, not if you're trying to sample.

Reply:
This max is over the auxiliary variables in the four corner tiles, not over a distribution. So we're checking if the aux var in the current tile is the highest of the four, and if so turning the other ones off. I'll add some better documentation describing what's going on here.

Reply:
Isn't it suppose to be probabilistic? i.e. the tile with the maximum probability isn't always the one selected to explain the corner, it's just the most frequent.

Reply:
`suppress_tiles()` is supposed to get a point estimate of the auxiliary variables (a sample if coming from `sample()`, or the mode if from `variational_mode()`). So it is probabilistic, it's just being sampled before this point.

That hasn't fully been implemented yet (there's a todo in `sample()`) - so right now it is deterministic. I'll update the documentation and add that in too.